federation/README.md

```diff
@@ -0,0 +1,61 @@
+# Cluster Federation
+
+Kubernetes Cluster Federation enables users to federate multiple
+Kubernetes clusters. Please see the [user guide](http://kubernetes.io/docs/admin/federation/)
+and the [admin guide](http://kubernetes.io/docs/user-guide/federation/federated-services/)
+for more details about setting up and using the Cluster Federation.
+
+# Building Kubernetes Cluster Federation
+
+Please see the [Kubernetes Development Guide](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/development.md)
+for initial setup. Once you have the development environment setup
+as explained in that guide, you also need to install [`jq`](https://stedolan.github.io/jq/download/)
```

Comment:
Thanks for starting this conversation. I meant to debate about this but somehow missed it while sending this PR.

My argument to myself so far has been that:

1. `jq` is really easy to install. It is in the upstream repositories of popular Linux distributions. No additional repositories needed, just `apt-get install jq`, `dnf install jq` and so on. There are packages and/or binaries readily available for Mac, Windows and other platforms - https://stedolan.github.io/jq/download/. So it is not that much of a hassle IMO.
2. This is only dev environments. We will not require `jq` in the shipped release/distribution.
3. It is not doing something special that we cannot accomplish with `sed`, but the intent is so much clear with `jq`. Also, it guarantees that it modifies only that particular JSON object we intend to modify and not accidentally modify some other field which matches the regexp.

Please let me know your thoughts. Open for suggestions.